runtime/cfdumper/main.c

```diff
@@ -6656,54 +6653,58 @@ static void j9_formatBytecodes(J9ROMClass* romClass, J9ROMMethod* method, U_8* b
 }
 
 
-
-static I_32 processDirectory(char* dir, BOOLEAN recursive, U_32 flags)
+static I_32
+processDirectory(char *dir, BOOLEAN recursive, U_32 flags)
 {
-	char *pathBuffer;
-	char *resultBuffer;
-	UDATA handle;
-	IDATA result, length, length2;
+	char *pathBuffer = NULL;
+	char *resultBuffer = NULL;
+	UDATA handle = 0;
+	IDATA result = 0;
+	IDATA length = 0;
 
 	PORT_ACCESS_FROM_PORT(portLib);
 
 	pathBuffer = j9mem_allocate_memory(EsMaxPath, J9MEM_CATEGORY_CLASSES);
-	if(pathBuffer == NULL) return RET_ALLOCATE_FAILED;
+	if (NULL == pathBuffer) {
+		return RET_ALLOCATE_FAILED;
+	}
+
 	resultBuffer = j9mem_allocate_memory(EsMaxPath, J9MEM_CATEGORY_CLASSES);
-	if(resultBuffer == NULL)
-	{
+	if (NULL == resultBuffer) {
 		j9mem_free_memory(pathBuffer);
 		return RET_ALLOCATE_FAILED;
 	}
+
 	length = strlen(dir);
-	strncpy(resultBuffer, dir, length);
+	if (length >= EsMaxPath) {
+		j9tty_printf(PORTLIB, "Could not open directory %s (path too long)\n", dir);
+		goto cleanup;
+	}
+	memcpy(resultBuffer, dir, length + 1);
 
 	handle = j9file_findfirst(dir, pathBuffer);
-	if(handle == (UDATA)-1)
-	{
+	if (~(UDATA)0 == handle) {
 		j9tty_printf(PORTLIB, "Could not open directory %s\n", dir);
 		goto cleanup;
 	}
+
 	result = handle;
-	while(result != -1)
-	{
-		if(!(pathBuffer[0] == '.' && (pathBuffer[1] == '\0' || (pathBuffer[1] == '.' && pathBuffer[2] == '\0'))))
-		{
+	while (-1 != result) {
+		/* ignore "." and ".." */
+		if ((0 != strcmp(pathBuffer, ".")) && (0 != strcmp(pathBuffer, ".."))) {
+			IDATA length2 = length + strlen(pathBuffer);
+			if (length2 + 1 >= EsMaxPath) {
+				j9tty_printf(PORTLIB, "Could not open %s%s (path too long)\n", dir, pathBuffer);
+				break;
+			}
 			strcpy(resultBuffer + length, pathBuffer);
-			length2 = strlen(resultBuffer);
-			if(recursive && (j9file_attr(resultBuffer) == EsIsDir))
-			{
-#if defined(WIN32) || defined(OS2)
-				resultBuffer[length2] = '\\';
-#else
-				resultBuffer[length2] = '/';
-#endif
+
+			if (recursive && (j9file_attr(resultBuffer) == EsIsDir)) {
+				resultBuffer[length2] = PATH_SEP_CHAR;
 				resultBuffer[length2 + 1] = '\0';
 				processDirectory(resultBuffer, TRUE, flags);
-			}
-			else
-			{
-				if((length2 > 6) && !strcmp(resultBuffer + length2 - 6, ".class"))
-				{
+			} else {
+				if ((length2 > 6) && 0 == strcmp(resultBuffer + length2 - 6, ".class")) {
```

Comment:
missing brackets around the second expression

Reply:
Fixed.